cmake/GenerateVersion.cmake

```diff
@@ -8,6 +8,14 @@ execute_process(
   ERROR_QUIET
   RESULT_VARIABLE RES)
 
+# Extract build info: "145+g0f9113907" => "g0f9113907"
+#   if on a "dirty" branch, this will also append `-dirty` to the end: "g0f9113907-dirty"
+string(REGEX REPLACE "[0-9]+\\+" "" NVIM_VERSION_BUILD "${GIT_TAG}")
+# Prefix with a `g`
+set(NVIM_VERSION_BUILD "g${NVIM_VERSION_BUILD}")
```

Comment:
`git describe` already adds the "g".

If we do the replace here, the `+` doesn't exist. Need something like this instead:

```
Extract build info: "dev-145-g0f9113907" => "g0f9113907"
string(REGEX REPLACE "[^-]+$" "" NVIM_VERSION_BUILD "${GIT_TAG}")
```

Reply:
For me `git describe` doesn't suffix with a `g`:
```zsh
$ git --git-dir=./.git --work-tree=./ describe --first-parent --dirty --always
22996137c
```

I like the idea of only having `dirty` in for `NVIM_VERSION_BUILD` when the work-tree is dirty with a regex, very nice. I've updated to the regex and removed the prefix operation.